### PR TITLE
fix #235, don't allow duplicate tags

### LINKED ISSUE #235: Duplicate tags | You can enter the same tag twice

### PR DESCRIPTION
**Pull request by ad1992**
@mrdannael @prakhar1989  please look into this.

### CODE REVIEW

File: lib/ReactTags.js [hunk 1]
```diff
@@ -95,9 +95,9 @@ class ReactTags extends Component {
   handleDelete(i, e) {
     this.props.handleDelete(i, e);
     if (!this.props.resetInputOnDelete) {
-        this.textInput.focus();
```
Comment (prakhar1989) [inline lib/ReactTags.js]: Would recommend running this through prettier `npm run format` to make sure we are consistent formatting / linting wise.

File: lib/ReactTags.js [hunk 2]
```diff
@@ -244,6 +244,13 @@ class ReactTags extends Component {
   }
 
   addTag(tag) {
+    const { tags } = this.props;
+    const existingTags = tags.map(tag => tag.text.toLowerCase());
+
+    // Return if tag has been already added
+    if (existingTags.indexOf(tag.toLowerCase()) >= 0) {
```
Comment (prakhar1989) [inline lib/ReactTags.js]: How about using includes? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

File: test/reactTags.test.js [hunk 3]
```diff
@@ -155,6 +153,22 @@ test("handles the paste event and splits the clipboard on delimiters", () => {
...
 
+test("should not allow duplicate tags", () => {
+  const actual = [];
+  const $el = mount(
+    mockItem({
+      handleAddition(tag) {
+        actual.push(tag);
+      },
+    })
+  );
```
Comment (prakhar1989) [inline test/reactTags.test.js]: Can you add an assertion for checking if 'Apple' is already there?

File: lib/ReactTags.js (hunk 2 above)
Comment (ad1992) [reply lib/ReactTags.js]: @prakhar1989 since includes doesn't support IE thats why I used indexOf

File: lib/ReactTags.js (hunk 1 above)
Comment (ad1992) [reply lib/ReactTags.js]: I ran `npm run format` which resulted in this change. Might be  `npm run format ` was not run when the above code was pushed.

File: test/reactTags.test.js (hunk 3 above)
Comment (ad1992) [reply test/reactTags.test.js]: added 👍

**Comment by prakhar1989** [comment]
Looks great, thanks for making this change Aakansha 👍 

LGTM from my side, but I'd recommend waiting for @mrdannael to take a quick look.

**Comment by mrdannael** [comment]
Looks good to me :)

**Comment by ad1992** [comment]
@prakhar1989 @mrdannael can this pr be merged?

**Comment by mrdannael** [comment]
Merged :+1:

**Comment by ad1992** [comment]
@mrdannael can you also update the link http://prakhar1989.github.io/react-tags/example/ so it does't add duplicate tags

**Comment by mrdannael** [comment]
I'll fix it with another issue.

**Pull request merged**